src/som/vmobjects/SClass.java

```diff
@@ -204,6 +204,16 @@ public SClass getClassCorrespondingTo(final MixinDefinitionId mixinId) {
     return cls;
   }
 
+  public SClass getClassWithMixinCorrespondingTo(final MixinDefinitionId mixinId) {
```

Comment:
hmm, I don't like the name.
If I ask `clazz.getClass*()`, I think I expect the class of the class with some property.
How about just `getMixin(..)`?

Also see my comment for `getClassCorrespondingTo()`

Reply:
`getClassWithMixinCorrespondingTo` returns the receiver class or one of its superclasses. The class returned is that one who has had the given mixin applied. For example if we have an class `C` defined as:

```
A = class()()
B = class Object <: A ()()
C = class B ()()
```

Then the result of `C. getClassWithMixinCorrespondingTo(A)` is `B`. Does that make sense? Do you have a suggestion for the method name, given this example?

Reply:
Wait, let's just talk about the java level and the naming, not the functionality on the language level.

The first thing that `getClassWithMixinCorrespondingTo(.)` can return is `this` (line 210).
The name of the method does not make me expect that.

If I ask any object in Java `this.getClass()`, I'll never get `this` returned, right?